### PR TITLE
fixed collapseIntervals error on null intervals

### PR DESCRIPTION
**Pull request by natjoe4**
Pull requests into cql-execution require the following.
Submitter and reviewer should ✔ when done.
For items that are not-applicable, mark "N/A" and ✔.

[CDS Connect](https://cds.ahrq.gov/cdsconnect) and [Bonnie](https://github.com/projecttacoma/bonnie) are the main users of this repository.
It is strongly recommended to include a person from each of those projects as a reviewer.

Submitted in conjunction with @elsaperelli
## Summary
`collapseIntervals` function previously threw `intervals is not iterable` error when `intervals` argument was null. Moved the `intervals == null` check to catch null values before entering the `for` loop. Restored existing testing for this case. Fixes [fqm-execution issue 227]( https://github.com/projecttacoma/fqm-execution/issues/227).

**Submitter:**
- [✔] This pull request describes why these changes were made
- [✔] Code diff has been done and been reviewed (it does not contain: additional white space, not applicable code changes, debug statements, etc.)
- [✔] Tests are included and test edge cases
- [✔] Tests have been run locally and pass
- [✔] Code coverage has not gone down and all code touched or added is covered.
- [✔] Code passes lint and prettier (hint: use `npm run test:plus` to run tests, lint, and prettier)
- [✔] All dependent libraries are appropriately updated or have a corresponding PR related to this change
- [✔] `cql4browsers.js` built with `npm run build:browserify` if source changed.

**Reviewer:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

File: test/elm/interval/data.js [hunk 1]
```diff
@@ -102774,15 +102774,35 @@ define CollapseQuantityIntervalListWithNullHighNoOverlap: collapse QuantityInter
...
+         "libraryId" : "TestSnippet",
+         "libraryVersion" : "1",
+         "startLine" : 71,
+         "startChar" : 22,
+         "endLine" : 71,
+         "endChar" : 34,
+         "message" : "Call to operator Collapse(System.Any,System.Quantity) is ambiguous with: \n  - Collapse(list<interval<System.Integer>>,System.Quantity)\n  - Collapse(list<interval<System.Quantity>>,System.Quantity)\n  - Collapse(list<interval<System.Date>>,System.Quantity)\n  - Collapse(list<interval<System.DateTime>>,System.Quantity)\n  - Collapse(list<interval<System.Time>>,System.Quantity)",
+         "errorType" : "semantic",
+         "errorSeverity" : "error",
+         "type" : "CqlToElmError"
```
Comment (cmoesel) [inline test/elm/interval/data.js:102801]: Note that the Translation error is still here -- so the output ELM is not what you think it is.

File: test/elm/interval/data.js [hunk 2]
```diff
@@ -106623,84 +106643,108 @@ module.exports['Collapse'] = {
...
+                        "r" : "389",
+                        "value" : [ "collapse ","null" ]
+                     } ]
+                  } ]
+               }
+            } ],
+            "expression" : {
+               "localId" : "390",
+               "type" : "Null"
+            }
```
Comment (cmoesel) [inline test/elm/interval/data.js:106668]: Due to the translation error, the ELM is actually just a literal `Null` -- so it's not actually testing the `Collapse` code. I think we can fix this in the test CQL via a cast.

**Review by cmoesel — APPROVED** [review]
Looks good to me! Thanks again for this!
